jt_notes.md

```diff
@@ -0,0 +1,127 @@
+**I [jonny] have made changes to the source code on a new branch (todos, improved comments, and a few changes to function/variable names etc)**. Where possible I have edited as I go, but am hoping for your input on resolving the queries in the todos. The more important conceptual queries are listed below. **I have also included a preliminary sketch of the architecture I have in mind**, although this is currently limited by my understanding of how the flask stuff works, particularly around multithreading, since that will have a big influence on what a correct and best way to handle all this in python will be. My biggest hangup is that I cannot see any connection between app.py and camera_pi.py (and in particular the frames() iterator). I therefore cannot see how the Flask app stuff is meant to work when running live on a RPi with the PiCam.
+
+------------------------------------ **Queries to Chas about the different modules ------------------------------------**
+
+cpi.py
+
+OpticalGater class that extends picamera.array.PiYUVAnalysis. It appears to maintain history, e.g. frame_history, period_history.
+
+**For chas:** my opinion is frame_history is really crying out for a class container, or at least an array with named columns (or a list-of-dictionaries, though I'm less keen on that). At present it requires the developer to keep track of the meaning of each column, and the format not easily extensible.
+
+**For chas:** load_data() - "Place holder function for loading data in emulator". What is the intention of this, and is there a clear idea of what an emulator would look like?
+
+**For me:** this is a general question for Chas: how does he intend the emulator to work? I don't think I've fully followed his emulator stuff in "app". Might need to talk through that one together...
```

Comment:
In my ideal future, the code will be able to stream frames from an open  file source just as it streams from a camera source, e.g. https://github.com/jrosebr1/imutils/blob/master/imutils/video/filevideostream.py